### PR TITLE
refactor(crux-ui): extend eslint

### PR DESCRIPTION
**Pull request by balanceee**
Add airbnb's eslint config as default to our config

### CODE REVIEW

File: web/crux-ui/src/components/products/versions/images/select-images-card.tsx [hunk 1]
```diff
@@ -52,48 +53,46 @@ const SelectImagesCard = (props: SelectImagesCardProps) => {
...
-  sock.on(WS_TYPE_DYO_ERROR, (message: DyoApiError) => {
+  sock.on(WS_TYPE_DYO_ERROR, (_: DyoApiError) => {
     setSearching(false)
   })
 
   useEffect(() => setRegistry(registriesFound ? registries[0] : null), [registries, registriesFound])
 
-  const findImage = (registry: Registry, filter?: string) => {
-    if (!registry || filter.length < IMAGE_FILTER_MIN_LENGTH) {
+  const findImage = (registryArg: Registry, filterArg?: string) => {
```
Comment (m8vago) [inline web/crux-ui/src/components/products/versions/images/select-images-card.tsx:70]: reg, filterText

File: web/crux-ui/src/components/products/versions/version-sections.tsx [hunk 2]
```diff
@@ -56,18 +70,16 @@ const VersionSections = (props: VersionSectionsProps) => {
   }
 
   const onAddDeployment = async (deploymentId: string) =>
-    router.push(deploymentUrl(props.product.id, props.version.id, deploymentId))
+    router.push(deploymentUrl(product.id, version.id, deploymentId))
 
-  const onReorderImages = (images: VersionImage[]) => {
-    const ids = images.map(it => it.id)
+  const onReorderImages = (imagesArg: VersionImage[]) => {
```
Comment (m8vago) [inline web/crux-ui/src/components/products/versions/version-sections.tsx:75]: no arg postfix ples

File: web/crux-ui/src/components/registries/registry-card.tsx [hunk 3]
```diff
@@ -17,38 +17,36 @@ interface RegistryCardProps extends Omit<DyoCardProps, 'children'> {
...
-          buttonClassName="ml-auto"
-          modalTitle={registry.name}
-        />
-
-        <RegistryTypeTag className="ml-auto mr-4" type={registry.type} />
-      </DyoCard>
-    </>
+    <DyoCard className={clsx(className ?? 'p-6', 'flex flex-col')}>
+      <div className={clsx(onClick ? 'cursor-pointer' : null, 'flex flex-row')} onClick={onClick}>
+        {!registry.icon ? getDefaultImage() : <DyoBadge large icon={registry.icon} />}
```
Comment (m8vago) [inline web/crux-ui/src/components/registries/registry-card.tsx:29]: get rid of getDefaultImage

File: web/crux-ui/src/components/shared/key-value-input.tsx [hunk 4]
```diff
@@ -14,23 +72,17 @@ interface KeyValueInputProps {
...
 
-  const [state, dispatch] = useReducer(reducer, props.items)
+  const [state, dispatch] = useReducer(reducer, items)
 
-  const stateToElements = (items: UniqueKeyValue[]) => {
+  const stateToElements = (itemsArg: UniqueKeyValue[]) => {
     const result = []
 
-    items.forEach(item =>
+    itemsArg.forEach(item =>
```
Comment (m8vago) [inline web/crux-ui/src/components/shared/key-value-input.tsx:85]: no arg ples

File: web/crux-ui/src/components/shared/page-menu.tsx [hunk 5]
```diff
@@ -127,16 +137,18 @@ interface SaveDiscardPageMenuProps {
 }
 
 export const SaveDiscardPageMenu = (props: SaveDiscardPageMenuProps) => {
+  const { className, saveRef, onSave: propsOnSave, onDiscard } = props
+
   const { t } = useTranslation('common')
 
   const onSave = () => {
```
Comment (m8vago) [inline web/crux-ui/src/components/shared/page-menu.tsx:144]: onSaveClick, so we can get rid of propsOnSave

File: web/crux-ui/src/components/shared/sh-editor.tsx [hunk 6]
```diff
@@ -9,31 +9,29 @@ interface ShEditorProps {
...
       <Editor
-        readOnly={props.readOnly}
+        readOnly={readOnly}
         padding={2}
         tabSize={2}
         insertSpaces
         value={state}
         onValueChange={onChange}
-        highlight={value => highlight(value, languages['shell'], 'shell')}
+        highlight={valueArg => highlight(valueArg, languages.shell, 'shell')}
```
Comment (m8vago) [inline web/crux-ui/src/components/shared/sh-editor.tsx:34]: no arg

File: web/crux-ui/src/components/team/edit-team-card.tsx [hunk 7]
```diff
@@ -20,10 +21,12 @@ interface EditTeamCardProps {
 }
 
 const EditTeamCard = (props: EditTeamCardProps) => {
+  const { className, team: teamProps, onTeamEdited, submitRef } = props
+
   const { t } = useTranslation('teams')
 
   const [team, setTeam] = useState<Team>(
-    props.team ?? {
+    teamProps ?? {
```
Comment (m8vago) [inline web/crux-ui/src/components/team/edit-team-card.tsx:29]: propsTeam

File: web/crux-ui/src/components/team/user-role-action.tsx [hunk 8]
```diff
@@ -17,24 +17,25 @@ interface UserRoleActionProps {
...
-  const role = user.role
+  const { role } = user
 
   const [updating, setUpdating] = useState(false)
   const [roleUpdateModalConfig, confirmRoleUpdate] = useConfirmation()
 
   const handleApiError = defaultApiErrorHandler(t)
 
-  const onUpdateUserRole = (role: UserRole, promote: boolean) =>
+  const onUpdateUserRole = (roleArg: UserRole, promote: boolean) =>
```
Comment (m8vago) [inline web/crux-ui/src/components/team/user-role-action.tsx:31]: arg

File: web/crux-ui/src/const.ts [hunk 9]
```diff
@@ -24,4 +24,56 @@ export const INVITE_LINK_EXPIRATION = '12h'
 export const ATTRIB_CSRF = 'csrf_token'
 export const HEADER_SET_COOKIE = 'set-cookie'
 export const AUTH_RESEND_DELAY = 30 // seconds
-export const WEBOOK_TEST_DELAY = 500 //millis
+export const WEBOOK_TEST_DELAY = 500 // millis
+
+// Routes:
+export const ROUTE_INDEX = '/'
```
Comment (m8vago) [inline web/crux-ui/src/const.ts:30]: move back routes to routes.ts

File: web/crux-ui/src/elements/dyo-button.tsx [hunk 10]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/button-has-type */
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-button.tsx:1]: eslint-disable-nextline

File: web/crux-ui/src/elements/dyo-checkbox.tsx [hunk 11]
```diff
@@ -13,9 +13,9 @@ interface DyoCheckboxProps {
 const DyoCheckbox = (props: DyoCheckboxProps) => {
   const { className, name, checked, setFieldValue, onCheckedChange } = props
 
-  const handleCheckedChange = checked => {
-    setFieldValue?.call(this, name, checked, false)
-    onCheckedChange?.call(this, checked)
+  const handleCheckedChange = checkedArg => {
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-checkbox.tsx:16]: narg

File: web/crux-ui/src/elements/dyo-container.tsx [hunk 12]
```diff
@@ -5,13 +5,12 @@ export interface DyoContainerProps extends React.HTMLProps<HTMLDivElement> {
   component?: string
 }
 
-export const DyoContainer = ({ component = 'div', className, children, ...props }: DyoContainerProps) => {
-  return React.createElement(
+export const DyoContainer = ({ component = 'div', className, children, ...props }: DyoContainerProps) =>
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-container.tsx:8]: get rid of it from error.tsx and delete this file and header.tsx as well

File: web/crux-ui/src/elements/dyo-img-button.tsx [hunk 13]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/button-has-type */
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-img-button.tsx:1]: eslint-next-line etc.

File: web/crux-ui/src/elements/dyo-input.tsx [hunk 14]
```diff
@@ -11,28 +11,26 @@ export interface DyoInputProps extends React.InputHTMLAttributes<HTMLInputElemen
...
       {!label ? null : (
-        <DyoLabel className="mt-8 mb-2.5" htmlFor={props.name}>
+        <DyoLabel className="mt-8 mb-2.5" htmlFor={name}>
           {label}
         </DyoLabel>
       )}
 
       <input
-        {...forwaredProps}
+        {...props}
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-input.tsx:25]: investigate: why forwardedProps is not working

File: web/crux-ui/src/elements/dyo-modal.tsx [hunk 15]
```diff
@@ -17,35 +17,31 @@ export interface DyoModalProps {
...
-          {props.children}
-
-          <div className="mx-auto mt-auto pt-8">{buttons}</div>
-        </DyoCard>
-      </Dialog>
-    </>
+    <Dialog
+      className="flex fixed inset-0 bg-light-grey bg-opacity-50 h-screen w-screen"
+      open={open}
+      onClose={() => onClose()}
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-modal.tsx:30]: no need for the lambda

File: web/crux-ui/src/elements/dyo-switch.tsx [hunk 16]
```diff
@@ -10,12 +10,12 @@ interface DyoSwitchProps extends Omit<React.InputHTMLAttributes<HTMLInputElement
...
 }
 
-export const DyoSwitch = (props: DyoSwitchProps) => {
+const DyoSwitch = (props: DyoSwitchProps) => {
   const { fieldName, checked, setFieldValue, onCheckedChange } = props
 
-  const handleCheckedChange = checked => {
-    setFieldValue?.call(this, fieldName, checked, false)
-    onCheckedChange?.call(this, checked)
+  const handleCheckedChange = checkedArg => {
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-switch.tsx:16]: :(

File: web/crux-ui/src/elements/dyo-toggle.tsx [hunk 17]
```diff
@@ -11,18 +11,18 @@ interface DyoToggleProps extends Omit<React.InputHTMLAttributes<HTMLInputElement
...
 
-export const DyoToggle = (props: DyoToggleProps) => {
-  const { name, checked, setFieldValue, onCheckedChange } = props
+const DyoToggle = (props: DyoToggleProps) => {
+  const { name, checked, setFieldValue, onCheckedChange, className, nameUnchecked, nameChecked } = props
 
-  const handleCheckedChange = checked => {
-    setFieldValue?.call(this, name, checked, false)
-    onCheckedChange?.call(this, checked)
+  const handleCheckedChange = checkedArg => {
```
Comment (m8vago) [inline web/crux-ui/src/elements/dyo-toggle.tsx:17]: narg :)

File: web/crux-ui/src/models.ts [hunk 18]
```diff
@@ -657,6 +665,69 @@ export const roleToText = (role: UserRole) => {
   }
 }
 
+export const invalidArgument = (property: string, description?: string, value?: string): DyoApiError => ({
```
Comment (m8vago) [inline web/crux-ui/src/models.ts:668]: move it to an error-responses.ts

File: web/crux-ui/src/pages/api/nodes/[nodeId]/setup.ts [hunk 19]
```diff
@@ -1,9 +1,9 @@
 import { NodeType } from '@app/models'
-import { nodeType } from '@app/validation'
+import { nodeType as nodeTypeValidation } from '@app/validation'
```
Comment (m8vago) [inline web/crux-ui/src/pages/api/nodes/[nodeId]/setup.ts:2]: nodeTypeSchema or nodeTypeRule ?

File: web/crux-ui/src/pages/nodes/[nodeId].tsx [hunk 20]
```diff
@@ -19,21 +20,23 @@ interface NodeDetailsProps {
...
 
-  const [node, setNode] = useState(props.node)
+  const [node, setNode] = useState(propsNode)
   const [editing, setEditing] = useState(false)
   const submitRef = useRef<() => Promise<any>>()
 
   const handleApiError = defaultApiErrorHandler(t)
 
-  const onNodeEdited = (node: DyoNodeDetails, shouldClose?: boolean) => {
+  const onNodeEdited = (nodeArg: DyoNodeDetails, shouldClose?: boolean) => {
```
Comment (m8vago) [inline web/crux-ui/src/pages/nodes/[nodeId].tsx:35]: noarg

File: web/crux-ui/src/pages/notifications/[notificationId].tsx [hunk 21]
```diff
@@ -45,9 +48,9 @@ const NotificationDetailsPage = (props: NotificationDetailsPageProps) => {
     url: ROUTE_NOTIFICATIONS,
   }
 
-  const onSubmitted = (notification: NotificationDetails) => {
+  const onSubmitted = (notificationArg: NotificationDetails) => {
```
Comment (m8vago) [inline web/crux-ui/src/pages/notifications/[notificationId].tsx:51]: noarg

File: web/crux-ui/src/pages/products/[productId]/versions/[versionId]/deployments/[deploymentId].tsx [hunk 22]
```diff
@@ -136,8 +137,8 @@ const DeploymentDetailsPage = (props: DeploymentDetailsPageProps) => {
     onOpenLog()
   }
 
-  const onDeploymentEdited = deployment => {
-    setDeployment(deployment)
+  const onDeploymentEdited = deploymentArg => {
```
Comment (m8vago) [inline web/crux-ui/src/pages/products/[productId]/versions/[versionId]/deployments/[deploymentId].tsx:140]: noarg

File: web/crux-ui/src/pages/products/[productId]/versions/[versionId]/deployments/[deploymentId]/deploy.tsx [hunk 23]
```diff
@@ -41,19 +41,21 @@ interface DeployPageProps {
...
+  const [status, setStatus] = useState<DeploymentStatus>(propsDeployment.status)
 
   const deployment = {
-    ...props.deployment,
+    ...propsDeployment,
     status,
   }
-  const product = deployment.product
-  const version = deployment.version
+  const { product } = deployment
```
Comment (m8vago) [inline web/crux-ui/src/pages/products/[productId]/versions/[versionId]/deployments/[deploymentId]/deploy.tsx:57]: product, version

File: web/crux-ui/src/pages/registries/[registryId].tsx [hunk 24]
```diff
@@ -19,19 +20,21 @@ interface RegistryDetailsPageProps {
...
   const [editing, setEditing] = useState(false)
   const submitRef = useRef<() => Promise<any>>()
 
   const handleApiError = defaultApiErrorHandler(t)
 
-  const onRegistryEdited = registry => {
+  const onRegistryEdited = registryArg => {
     setEditing(false)
-    setRegistry(registry)
+    setRegistry(registryArg)
```
Comment (m8vago) [inline web/crux-ui/src/pages/registries/[registryId].tsx:37]: noarg

File: web/crux-ui/src/server/crux/mappers/deployment-mapper.ts [hunk 25]
```diff
@@ -0,0 +1,46 @@
+import { DeploymentEventType, DeploymentStatus, Instance } from '@app/models'
```
Comment (m8vago) [inline web/crux-ui/src/server/crux/mappers/deployment-mapper.ts:1]: mappers*

File: web/crux-ui/src/server/crux/mappers/health-mapper.ts [hunk 26]
```diff
@@ -0,0 +1,7 @@
+import { ServiceStatus } from '@app/models'
```
Comment (m8vago) [inline web/crux-ui/src/server/crux/mappers/health-mapper.ts:1]: mappers*

File: web/crux-ui/src/server/crux/mappers/node-mapper.ts [hunk 27]
```diff
@@ -0,0 +1,22 @@
+import { ContainerState, NodeStatus, NodeType, NODE_TYPE_VALUES } from '@app/models'
```
Comment (m8vago) [inline web/crux-ui/src/server/crux/mappers/node-mapper.ts:1]: mappers*

File: web/crux-ui/src/server/crux/mappers/notification-mapper.ts [hunk 28]
```diff
@@ -0,0 +1,12 @@
+import { NotificationType } from '@app/models'
```
Comment (m8vago) [inline web/crux-ui/src/server/crux/mappers/notification-mapper.ts:1]: mappers*

File: web/crux-ui/src/server/crux/mappers/product-mapper.ts [hunk 29]
```diff
@@ -0,0 +1,10 @@
+import { ProductType } from '@app/models'
```
Comment (m8vago) [inline web/crux-ui/src/server/crux/mappers/product-mapper.ts:1]: mappers*

File: web/crux-ui/src/server/crux/mappers/registry-mapper.ts [hunk 30]
```diff
@@ -0,0 +1,7 @@
+import { RegistryType } from '@app/models'
```
Comment (m8vago) [inline web/crux-ui/src/server/crux/mappers/registry-mapper.ts:1]: mappers*

File: web/crux-ui/src/server/crux/mappers/version-mapper.ts [hunk 31]
```diff
@@ -0,0 +1,11 @@
+import { VersionType } from '@app/models'
```
Comment (m8vago) [inline web/crux-ui/src/server/crux/mappers/version-mapper.ts:1]: mappers*

File: web/crux-ui/src/server/crux/team-service.ts [hunk 32]
```diff
@@ -29,17 +29,12 @@ import {
...
-  UserRole as ProtoUserRole,
-  userRoleFromJSON,
-  userRoleToJSON,
-  UserStatus as ProtoUserStatus,
-  userStatusToJSON,
 } from '@app/models/grpc/protobuf/proto/crux'
 import { Identity } from '@ory/kratos-client'
 import { protomisify } from '@server/crux/grpc-connection'
-import { DyoApiError } from '@server/error-middleware'
+/* eslint-disable import/no-cycle */
```
Comment (m8vago) [inline web/crux-ui/src/server/crux/team-service.ts:35]: let me know when you're ready to discuss this in person

File: web/crux-ui/src/server/middlewares.ts [hunk 33]
```diff
@@ -61,7 +63,6 @@ export const withMiddlewares = (
 
           if (!handler) {
             res.status(405).end()
-            return
```
Comment (m8vago) [inline web/crux-ui/src/server/middlewares.ts:64]: this is necessary

File: web/crux-ui/src/websockets/websocket-client.ts [hunk 34]
```diff
@@ -331,7 +211,7 @@ export class WebSocketClient {
           this.errorHandler(message.payload as DyoApiError)
         }
 
-        this.routes.forEach(route => route.endpoints.forEach(it => it.onMessage(message)))
+        this.routes.forEach(routeArg => routeArg.endpoints.forEach(it => it.onMessage(message)))
```
Comment (m8vago) [inline web/crux-ui/src/websockets/websocket-client.ts:214]: args

File: web/crux-ui/src/validation.ts [hunk 35]
```diff
@@ -71,7 +71,7 @@ const registryCredentialRole = yup
     is: (type, _private) =>
       type === 'gitlab' || type === 'github' || ((type === 'v2' || type === 'google') && _private),
     then: yup.string().required(),
-    otherwise: yup.mixed().transform(it => (it ? it : undefined)),
```
Comment (m8vago) [inline web/crux-ui/src/validation.ts:74]: @robot9706 isn't this was necessary?

File: web/crux-ui/hete.txt [hunk 36]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (m8vago) [inline web/crux-ui/hete.txt:1]: is this the real life?

File: web/crux-ui/src/components/main/team-selection-card.tsx [hunk 37]
```diff
@@ -34,40 +34,42 @@ const TeamSelectionCard = (props: TeamSelectionCardProps) => {
...
-                alt={t('teamAvatar')}
-                width={32}
-                height={32}
-              />
-              <div className="ml-4">{it.name}</div>
-            </div>,
-          ]
-          /* eslint-enable react/jsx-key */
-        }}
+        itemBuilder={it => itemTemplate(it)}
```
Comment (m8vago) [inline web/crux-ui/src/components/main/team-selection-card.tsx:72]: itemBuilder={itemTemplate}

File: web/crux-ui/src/components/products/versions/version-deployments-section.tsx [hunk 38]
```diff
@@ -89,21 +102,7 @@ const VersionDeploymentsSection = (props: VersionDeploymentsSectionProps) => {
...
-            </div>,
-            <div>{it.nodeName}</div>,
-            <div>{it.date}</div>,
-            <div>{it.prefix}</div>,
-            <DeploymentStatusTag className="w-fit m-auto" status={it.status} />,
-            <Image src="/deploy.svg" alt={t('common:deploy')} width={24} height={24} />,
-          ]
-          /* eslint-enable react/jsx-key */
-        }}
+        itemBuilder={it => itemTemplate(it)}
```
Comment (m8vago) [inline web/crux-ui/src/components/products/versions/version-deployments-section.tsx:105]: `itemBuilder={itemTemplate}`

File: web/crux-ui/src/components/shared/json-editor.tsx [hunk 39]
```diff
@@ -31,64 +58,36 @@ const _JsonEditor = <T,>(props: JsonEditorProps<T>) => {
...
         onValueChange={onChange}
-        highlight={value => highlight(value, languages['json'], 'json')}
+        highlight={newValue => highlight(newValue, languages.json, 'json')}
       />
     </div>
   )
 }
 
-export default _JsonEditor
+export default JsonEditor
```
Comment (m8vago) [inline web/crux-ui/src/components/shared/json-editor.tsx:86]: Rename it to JsonEditorInternal or something, cause we're using dynamic module on it

File: web/crux-ui/src/pages/audit-log.tsx [hunk 40]
```diff
@@ -115,21 +128,7 @@ const AuditLogPage = (props: AuditLogPageProps) => {
...
-                  <div className="font-semibold min-w-max pl-2">{it.identityName}</div>,
-                  <div className="min-w-max">{utcDateToLocale(it.date)}</div>,
-                  <div>{beautifyAuditLogEvent(it.event)}</div>,
-                  <div className="cursor-pointer max-w-4xl truncate" onClick={() => onShowInfoClick(it)}>
-                    {it.info}
-                  </div>,
-                ]
-                /* eslint-enable react/jsx-key */
-              }}
+              itemBuilder={it => itemTemplate(it)}
```
Comment (m8vago) [inline web/crux-ui/src/pages/audit-log.tsx:131]: `itemBuilder={itemTemplate}`

File: web/crux-ui/src/pages/deployments.tsx [hunk 41]
```diff
@@ -37,42 +38,41 @@ const DeploymentsPage = (props: DeploymentsPageProps) => {
...
-                  alt={t('common:deploy')}
-                  width={24}
-                  height={24}
-                  className="mr-8 cursor-pointer"
-                  onClick={() => router.push(deploymentUrl(it.productId, it.versionId, it.id))}
-                />,
-              ]
-              /* eslint-enable react/jsx-key */
-            }}
+            itemBuilder={it => itemTemplate(it)}
```
Comment (m8vago) [inline web/crux-ui/src/pages/deployments.tsx:75]: `itemBuilder={itemTemplate}`

File: web/crux-ui/src/pages/teams/[teamId].tsx [hunk 42]
```diff
@@ -166,36 +194,7 @@ const TeamDetailsPage = (props: TeamDetailsPageProps) => {
...
-                  alt={t('common:delete')}
-                  width={24}
-                  height={24}
-                  onClick={() => onDeleteUser(it)}
-                />
-              ),
-            ]
-            /* eslint-enable react/jsx-key */
-          }}
+          itemBuilder={it => itemTemplate(it)}
```
Comment (m8vago) [inline web/crux-ui/src/pages/teams/[teamId].tsx:197]: `itemBuilder={itemTemplate}`

**Review by m8vago — APPROVED** [review]

**Review by robot9706 — APPROVED** [review]

**Pull request merged**